### PR TITLE
feature: Alonzo data mapping

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
To support applications operating in the Alonzo-era, the GraphQL model needs extending.

# Proposed Solution
Adds collateral, redeemers, and scripts concepts, and extends the protocol params
model.

# Important Changes Introduced

### CODE REVIEW

File: packages/api-cardano-db-hasura/test/__snapshots__/transactions.query.test.ts.snap [hunk 1]
```diff
@@ -105,6 +105,150 @@ Object {
 }
 `;
 
+exports[`transactions Returns transactions by hashes - alonzo-qa 1`] = `
```
Comment (rhyslbw) [inline packages/api-cardano-db-hasura/test/__snapshots__/transactions.query.test.ts.snap:108]: @toaspzoo Snapshots used to capture full response

File: packages/api-cardano-db-hasura/test/__snapshots__/epochs.query.test.ts.snap [hunk 2]
```diff
@@ -112,25 +112,78 @@ Object {
 }
 `;
 
-exports[`epochs Includes protocol params in effect for the epoch 1`] = `
+exports[`epochs Includes protocol params in effect for the epoch - alonzo-qa 1`] = `
```
Comment (rhyslbw) [inline packages/api-cardano-db-hasura/test/__snapshots__/epochs.query.test.ts.snap:115]: @toaspzoo Snapshot

File: packages/api-cardano-db-hasura/test/util.ts [hunk 3]
```diff
@@ -11,19 +11,27 @@ import util from '@cardano-graphql/util'
 // import { Genesis } from '@src/graphql_types'
 // import { CardanoNodeClient } from '@src/CardanoNodeClient'
 
-const getLastConfiguredMajorVersion = (network: string) =>
-  require(`../../../config/network/${network}/cardano-node/config.json`)['LastKnownBlockVersion-Major']
+// const getLastConfiguredMajorVersion = (network: string) =>
```
Comment (rhyslbw) [inline packages/api-cardano-db-hasura/test/util.ts:14]: These arguments are are not required for the e2e client. I will tidy up in a separate commit prior to the production release

File: packages/api-cardano-db-hasura/test/__snapshots__/transactions.query.test.ts.snap [hunk 4]
```diff
@@ -254,8 +415,12 @@ Object {
           },
         },
       },
+      "redeemers": Array [],
+      "scriptSize": 0,
+      "scripts": Array [],
       "size": 220,
       "totalOutput": "4924799478660",
+      "validContract": true,
```
Comment (toaspzoo) [inline packages/api-cardano-db-hasura/test/__snapshots__/transactions.query.test.ts.snap:423]: @rhyslbw what does this field mean? And should it be true value here?

Comment (rhyslbw) [reply packages/api-cardano-db-hasura/test/__snapshots__/transactions.query.test.ts.snap:423]: image

https://github.com/input-output-hk/cardano-db-sync/blob/alonzo-purple-1.0/doc/schema.md#tx

Comment (toaspzoo) [reply packages/api-cardano-db-hasura/test/__snapshots__/transactions.query.test.ts.snap:423]: Thank you

File: packages/api-cardano-db-hasura/schema.graphql [hunk 5]
```diff
@@ -1122,6 +1122,7 @@ type Transaction {
   size: BigInt!
   totalOutput: String!
   includedAt: DateTime!
+  # True unless a contract is invalid
```
Comment (rhyslbw) [inline packages/api-cardano-db-hasura/schema.graphql:1125]: @toaspzoo

**Review by toaspzoo — APPROVED** [review]

**Pull request merged**